src/cdk/table/array-data-source.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DataSource} from './data-source';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Array-based data source that can be used to connect to a CollectionViewer. Will trigger an
+ * update to the CollectionViewer whenever a new data array is set.
+ */
+export class ArrayDataSource<T> implements DataSource<T> {
```

Comment:
Overall this isn't really what I had in mind for `ArrayDataSource`. I had been thinking that his class would capture all of the behavior for applying sorting, pagination, and potentially a substring search filter.

If we do this, the biggest thing to decide is how we access the row data for sorting / potentially filtering in a property-renaming compatible way.